cloudinit/sources/helpers/azure.py

```diff
@@ -317,9 +323,8 @@ def __init__(
 
         for attr in ("container_id", "instance_id", "incarnation"):
             if getattr(self, attr) is None:
-                msg = 'Missing %s in GoalState XML'
-                LOG.warning(msg, attr)
-                report_diagnostic_event(msg % (attr,))
+                msg = 'Missing %s in GoalState XML' % attr
+                report_diagnostic_event(msg, logger_func=LOG.warning)
                 raise InvalidGoalStateXMLException(msg)
```

Comment:
@OddBloke I think we missed the bug here the last time :) The error message being raised through the exception was incomplete and was not formatted with the missing expected attribute.